src/services/CompareUtils.jl

```diff
@@ -267,7 +267,7 @@ function compareFactor(A::DFGFactor,
   end
   @debug "compareFactor 4/5" TP
   if !(:varValsAll in skip) && hasfield(typeof(getSolverData(A).fnc), :varValsAll)
-    TP = TP & (skipcompute || compareAll(getSolverData(A).fnc.varValsAll, getSolverData(B).fnc.varValsAll, show=show, skip=skip))
+    TP = TP & (skipcompute || compareAll(getSolverData(A).fnc.varValsAll[], getSolverData(B).fnc.varValsAll[], show=show, skip=skip))
```

Comment:
should this not be done separately? Is it part of the IIF fixes?

Reply:
Good catch thanks, yes will do in separate PR as part of the IIF fixes.